client/src/components/form-viewer/view-form-field.tsx

```diff
@@ -0,0 +1,96 @@
+import React from 'react';
+import {
+  FormFieldSchema,
+  FormOptionDataSource,
+  FormOptionValue,
+} from '@konomi.ai/c-form';
+import { useFormContext } from 'react-hook-form';
+import { Typography } from '@mui/material';
+
+import { FormInputField } from '../form-inputs/FormInputField';
+import { FormSelect } from '../form-inputs/FormSelect';
+import { MultiSelect } from '../form-inputs/MultiSelect';
+
+type FormFieldProps = Omit<FormFieldSchema, 'id'>;
+
+function isOptionValue(
+  object: FormOptionValue | FormOptionDataSource,
+): object is FormOptionValue {
+  return 'id' in object && 'label' in object;
+}
+
+const isOptionValueArray = (options: unknown): options is FormOptionValue[] =>
+  Array.isArray(options) && options.every((option) => isOptionValue(option));
+
+export const ViewFormField = ({
+  name,
+  type,
+  optional,
+  options,
+  description,
+}: FormFieldProps) => {
+  const { control, setValue } = useFormContext();
+
+  switch (type) {
+    case 'text':
+      return (
+        <FormInputField
+          name={name}
+          control={control}
+          rules={{ required: !optional }}
+          label={description || name}
```

Comment:
The form label should just be the name itself. The label should never be the description.

We can enhance these field components to support help text which will allow us to render this information. I can log that as a follow up.

Also the preferred syntax for coalesce is `??` and not `||`